creme/neighbors/base_neighbors.py

```diff
@@ -0,0 +1,156 @@
+from collections import deque
+
+import numpy as np
+from sklearn.neighbors import KDTree
```

Comment:
Would it be possible to use the KD-tree from `scipy`? I would prefer not having `sklearn` as a dependency, if possible :)

Reply:
That's a good point. Until yesterday I didn't know scipy also had its own implementation. I'll check that. I agree with you about depending on `sklearn`: as far as I am concerned we use it in too few locations to make it a dependency.

Any suggestions here @jacobmontiel?

Reply:
Also worth considering, there is a Cython implementation in `scipy` (the implementation in `sklearn` is also in Cython):

https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.cKDTree.html#scipy.spatial.cKDTree

Switching implementations seems trivial as the interface is quite similar. However, the `sklearn` implementation seems to provide more features, for example it supports multiple distance metrics which is nice for the user.

Reply:
I did some reading earlier today on this subject. The `cKDTree` seems to only support the *Minkowski* distance and it seems good enough to me as we can get *Manhattan*, *Euclidean* and so on.

But one might want to experiment with different distances. What do you guys think?

Reply:
Now the algorithms are using the scipy version of KDTree (cKDTree).

Reply:
I added that possibility in the last commit. These hyperparameters are related to the KDTree construction. There is yet another set of useful parameters we could consider. They are used to make queries in the KDTree.